### DISCUSSION TITLE
machine learning only for certain albums

### DISCUSSION THREAD

**Discussion by crKRoEYS** [ente-io/ente#3189] [Enhancements]
Hi, since machine learning involves downloading the original data, which takes up a lot of local storage, I would find it important that this feature could be restricted to individual albums.

**Comment by ua741** [comment]
You can do initial indexing on desktop. After that, mobile client will only do indexing for newer files.

**Comment by crKRoEYS** [comment]
Could you explain that please? 1 Device do initial indexing, which is synced with all my devices, so the Indexes are in the Cloud? Because it needs local storage only on one device? If not, the pictures and videos have to upload again? And when indexing the newer files after a while needs too much local storage on my smartphone? Thanks! :)

**Reply by ua741** [reply]
Yes, indexes are stored on the cloud, e2ee. 
Currently, on mobile, for each file, we check on the client if the index exists or not. If it doesn't exist, the client will download the file (if it's not already present on the device), and index it. In any case, indexing is not related to uploading of pictures or videos.

Suppose you have indexed all your photos using either mobile or desktop (desktop will be faster for initial indexing, unless all of your photos are also present on the mobile). Now, the mobile client will need local storage only for downloading the models and the indexes.

> And when indexing the newer files after a while needs too much local storage on my smartphone
If the file is already present on the mobile, you don't need additional storage as we will reuse the local file for indexing.
